stacks-common/src/util/secp256r1.rs

```diff
@@ -0,0 +1,584 @@
+        }
+
+        let signature: P256Signature = self.key.sign(data_hash);
+        Ok(MessageSignature::from_p256_signature(&signature))
+    }
+}
+
+fn secp256r1_pubkey_serialize<S: serde::Serializer>(
+    pubk: &P256VerifyingKey,
+    s: S,
+) -> Result<S::Ok, S::Error> {
+    let public_key = P256PublicKey::from(pubk);
+    let encoded_point = public_key.to_encoded_point(true); // always serialize as compressed
+    let key_hex = to_hex(encoded_point.as_bytes());
+    s.serialize_str(key_hex.as_str())
+}
+
+fn secp256r1_pubkey_deserialize<'de, D: serde::Deserializer<'de>>(
+    d: D,
+) -> Result<P256VerifyingKey, D::Error> {
+    let key_hex = String::deserialize(d)?;
+    let key_bytes = hex_bytes(&key_hex).map_err(de_Error::custom)?;
+
+    let encoded_point = EncodedPoint::from_bytes(&key_bytes).map_err(de_Error::custom)?;
+    let public_key =
+        Option::<P256PublicKey>::from(P256PublicKey::from_encoded_point(&encoded_point))
+            .ok_or_else(|| de_Error::custom("Invalid public key"))?;
+    Ok(P256VerifyingKey::from(public_key))
+}
+
+fn secp256r1_privkey_serialize<S: serde::Serializer>(
+    privk: &P256SigningKey,
+    s: S,
+) -> Result<S::Ok, S::Error> {
+    let key_hex = to_hex(privk.to_bytes().as_slice());
+    s.serialize_str(key_hex.as_str())
+}
+
+fn secp256r1_privkey_deserialize<'de, D: serde::Deserializer<'de>>(
+    d: D,
+) -> Result<P256SigningKey, D::Error> {
+    let key_hex = String::deserialize(d)?;
+    let key_bytes = hex_bytes(&key_hex).map_err(de_Error::custom)?;
+
+    if key_bytes.len() != 32 {
+        return Err(de_Error::custom("Private key must be 32 bytes"));
+    }
+
+    let mut key_array = [0u8; 32];
+    key_array.copy_from_slice(&key_bytes);
+
+    let secret_key =
+        P256SecretKey::from_bytes(&GenericArray::from(key_array)).map_err(de_Error::custom)?;
+
+    Ok(P256SigningKey::from(secret_key))
+}
+
+/// Verify a secp256r1 signature.
+/// The message must be a 32-byte hash.
+/// The signature must be a 64-byte compact signature
+pub fn secp256r1_verify(
+    message_arr: &[u8],
+    signature_arr: &[u8],
+    pubkey_arr: &[u8],
+) -> Result<(), Secp256r1Error> {
+    if message_arr.len() != 32 {
+        return Err(Secp256r1Error::InvalidMessage);
+    }
+
+    if signature_arr.len() != 64 {
+        return Err(Secp256r1Error::InvalidSignature);
+    }
+
+    let encoded_point =
+        EncodedPoint::from_bytes(pubkey_arr).map_err(|_| Secp256r1Error::InvalidKey)?;
+
+    let public_key =
+        Option::<P256PublicKey>::from(P256PublicKey::from_encoded_point(&encoded_point))
+            .ok_or(Secp256r1Error::InvalidKey)?;
+    let verifying_key = P256VerifyingKey::from(public_key);
+
+    let signature =
+        P256Signature::from_slice(signature_arr).map_err(|_| Secp256r1Error::InvalidSignature)?;
+
+    verifying_key
+        .verify(message_arr, &signature)
+        .map_err(|_| Secp256r1Error::InvalidSignature)
```

Comment:
IMO it would be better to re-use our "library wrapper" code to prevent us from having differing applications
```suggestion
    let pk = Secp256r1PublicKey::from_slice(pubkey_arr).map_err(|_| Secp256r1Error::InvalidKey)?;
    let sig =
        MessageSignature::from_bytes(signature_arr).ok_or(Secp256r1Error::InvalidSignature)?;
    pk.verify_digest(message_arr, &sig)
        .map_err(|_| Secp256r1Error::InvalidSignature)?;
    Ok(())
```